.github/workflows/dasboard-format.yaml

```diff
@@ -0,0 +1,43 @@
+name: KEEP Token Dashboard Code Format Checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+defaults:
+  run:
+    working-directory: ./solidity/dashboard
+
+jobs:
+  dashboard-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      dashboard: ${{ steps.filter.outputs.dashboard }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          dashboard:
+            - './solidity/dashboard/**'
+
+  dashboard-code-format:
+    needs: dashboard-detect-changes
+    if: ${{ needs.dashboard-detect-changes.outputs.dashboard == 'true' }}
```

Comment:
Three comments regarding this section:
1. I would like to keep the implementation of different workflows quite similar, when possible (it makes it easier to do bulk edits when some changes are needed across different workflows), hence I would suggest changing name of the filter from `dashboard` to `path-filter`, as we call it in other workflows (although I see why you picked the current name and if you feel it's better, I'm ok with keeping it).
2. I don't remember the exact details now, but I believe that there was some issue with using `dorny/paths-filter` action for workflows started by `push` event (if I recall correctly, the problem with correct filtering could occur if feature branch was ahead of main certain amount of commits). That's why in other workflows we use the action only for workflows started by `pull-request` event. I think we need to do the same here.
3. I believe it's almost always a good idea to add `workflow_dispatch` trigger to the workflows.

```suggestion
on:
  push:
    branches:
      - main
    paths:
      - "solidity/dashboard/**"
  pull_request:
  workflow_dispatch:

defaults:
  run:
    working-directory: ./solidity/dashboard

jobs:
  dashboard-detect-changes:
    runs-on: ubuntu-latest
    outputs:
      dashboard: ${{ steps.filter.outputs.dashboard }}
    steps:
    - uses: actions/checkout@v2
      if: github.event_name == 'pull_request'

    - uses: dorny/paths-filter@v2
      if: github.event_name == 'pull_request'
      id: filter
      with:
        filters: |
          path-filter:
            - './solidity/dashboard/**'

  dashboard-code-format:
    needs: dashboard-detect-changes
    if: |
      github.event_name != 'pull_request'
        || needs.dashboard-detect-changes.outputs.path-filter == 'true'
```

Reply:
In this case, I think it makes sense to put `if` statement on the job level instead of for each `step`. 

```
jobs:
  dashboard-detect-changes:
    if: github.event_name == 'pull_request'
```

Thoughts?

Reply:
Normally I would agree, but in this particular job we need to keep the `if`s on the step level, as we want the `dashboard-detect-changes` job (but not it's step) to execute for workflow_dispatch and pull events. If we had the `if` on the job level, the job wouldn't be skipped for this type of events, resulting in skipping of jobs that have it configured as a dependency in the `needs` field.

But now when I think about it, maybe we can completely get rid of the `dashboard-detect-changes` job... This job is needed in the workflow if we have a situation where we want to make a passing of some job (like `dashboard-code-format`) mandatory for the merge of PR and when we want that job to be executed only if the PR touches the files in some specific location. I'm not sure if the `dashboard-code-format` satisfies that first condition.

@nkuba, will we be adding the `dashboard-code-format` as mandatory for PR merge? I can see that in `keep-ecdsa` we don't have the formatting job set as required...

Reply:
We should set the formatting job as required for PR merge.

Reply:
Ok, so then we need to keep the solution with `dashboard-detect-changes` job. @r-czajkowski please move the `if`s back to the step level.